cub/cub/device/dispatch/tuning/tuning_three_way_partition.cuh

```diff
@@ -0,0 +1,246 @@
+{
+  static constexpr int threads = 128;
+  static constexpr int items   = 7;
+
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_TRANSPOSE;
+
+  using delay_constructor = detail::no_delay_constructor_t<1040>;
+};
+
+template <class Input, class OffsetT>
+struct sm90_tuning<Input, OffsetT, input_size::_1, offset_size::_8>
+{
+  static constexpr int threads = 256;
+  static constexpr int items   = 24;
+
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_DIRECT;
+
+  using delay_constructor = detail::fixed_delay_constructor_t<4, 285>;
+};
+
+template <class Input, class OffsetT>
+struct sm90_tuning<Input, OffsetT, input_size::_2, offset_size::_8>
+{
+  static constexpr int threads = 640;
+  static constexpr int items   = 24;
+
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_TRANSPOSE;
+
+  using delay_constructor = detail::no_delay_constructor_t<245>;
+};
+
+template <class Input, class OffsetT>
+struct sm90_tuning<Input, OffsetT, input_size::_4, offset_size::_8>
+{
+  static constexpr int threads = 256;
+  static constexpr int items   = 23;
+
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_TRANSPOSE;
+
+  using delay_constructor = detail::no_delay_constructor_t<910>;
+};
+
+template <class Input, class OffsetT>
+struct sm90_tuning<Input, OffsetT, input_size::_8, offset_size::_8>
+{
+  static constexpr int threads = 256;
+  static constexpr int items   = 18;
+
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_TRANSPOSE;
+
+  using delay_constructor = detail::no_delay_constructor_t<1145>;
+};
+
+template <class Input, class OffsetT>
+struct sm90_tuning<Input, OffsetT, input_size::_16, offset_size::_8>
+{
+  static constexpr int threads = 256;
+  static constexpr int items   = 11;
+
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_TRANSPOSE;
+
+  using delay_constructor = detail::no_delay_constructor_t<1050>;
+};
+
+} // namespace three_way_partition
+
+template <class InputT, class OffsetT>
+struct device_three_way_partition_policy_hub
+{
+  /// SM35
+  struct Policy350 : ChainedPolicy<350, Policy350, Policy350>
+  {
+    constexpr static int ITEMS_PER_THREAD = Nominal4BItemsToItems<InputT>(9);
+
+    using ThreeWayPartitionPolicy = cub::AgentThreeWayPartitionPolicy<256,
+                                                                      ITEMS_PER_THREAD,
+                                                                      cub::BLOCK_LOAD_DIRECT,
+                                                                      cub::LOAD_DEFAULT,
+                                                                      cub::BLOCK_SCAN_WARP_SCANS>;
+  };
+
+  /// SM90
+  struct Policy900 : ChainedPolicy<900, Policy900, Policy350>
+  {
+    using tuning = detail::three_way_partition::sm90_tuning<InputT, OffsetT>;
+
+    using ThreeWayPartitionPolicy =
+      AgentThreeWayPartitionPolicy<tuning::threads,
```

Comment:
No Change requested: Why is this not qualified with cub::

Reply:
This code is in cub namespace and I think there's no strict policy on qualifying all cub entities at the moment.

Reply:
The question was because right above it is qualified. We should be consistent here